vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc

```diff
@@ -221,6 +221,7 @@ void Uphold::DisconnectWallet() {
     [](ledger::Result _){});
 
   ledger_->SaveExternalWallet(ledger::kWalletUphold, std::move(wallet));
+  ledger_->GetLedgerClient()->WalletDisconnected(ledger::kWalletUphold);
```

Comment:
Feels a little weird to punch a hole into the ledger client like this. It seemed like there was an `OnDisconnectWallet` observer. That can't be used for this purpose?

Reply:
I was wondering the same. The current design has us always access the client through `LedgerImpl` (not that I'm really a fan of everything going through `LedgerImpl`). Are we moving toward allowing internal classes access to ledger client?

Reply:
yup this is the plan. So that we can reduce functions in impl that are just proxies

Reply:
I will create PR tomorrow that will move all of this. But this PR looked like a good PR to start with it 😃